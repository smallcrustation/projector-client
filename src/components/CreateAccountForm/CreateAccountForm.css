form{
  display: flex;
  flex-direction: column;
}

input{
  width: 80%;
  border: 1px lightgray solid;
}

.error{
  color: red;
}