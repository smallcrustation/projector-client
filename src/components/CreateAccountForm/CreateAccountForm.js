import React from 'react'

import AuthApiService from '../../services/auth-api-service'
import Loading from '../Loading/Loading'

import './CreateAccountForm.css'

export default class Landing extends React.Component {
  state = { error: null, loading: false }

  handleSubmit = async e => {
    e.preventDefault()
    this.setState({ error: null, loading: true })

    const { username, email, pass, pass2 } = e.target

    // Validation... anymore make separate func
    let validErr = null
    if (pass.value !== pass2.value) {
      // remove return when more validation added
      validErr = 'Passwords must match!'
    }

    if (username.value.includes(' ')) {
      validErr
        ? (validErr += '\nUsername cannot contain spaces.')
        : (validErr = 'Username cannot contain spaces.')
    }

    if (validErr) {
      // console.log('valid err')
      this.setState({ loading: false })
      return this.setState({error: validErr})
    }

    const newUser = {
      username: username.value,
      email: email.value,
      password: pass.value
    }

    try {
      // eslint-disable-next-line no-unused-vars
      const savedUser = await AuthApiService.createAccount(newUser)
      username.value = ''
      email.value = ''
      pass.value = ''
      pass2.value = ''
      this.setState({ loading: false })
      this.props.onCreateAccountSuccess()
    } catch (err) {
      this.setState({ error: err.error, loading: false })
    }
  }

  render() {
    const { error, loading } = this.state
    return (
      <form className="CreateAccountForm" onSubmit={this.handleSubmit}>
        <div role="alert">{error && <p className="error">{error}</p>}</div>
        {loading ? (
          <Loading />
        ) : (
          <div>
            {' '}
            <div>
              <label htmlFor="username">Username</label>
              <input type="text" id="username" name="username" required />
            </div>
            <div>
              <label htmlFor="email">Email</label>
              <input type="email" id="email" name="email" required />
            </div>
            <div>
              <label htmlFor="pass">Password</label>
              <input type="password" id="pass" name="pass" required />
            </div>
            <div>
              <label htmlFor="pass2">Confirm Password</label>
              <input type="password" id="pass2" name="pass2" required />
            </div>
            <input type="submit" value="Create Account!" />
          </div>
        )}
      </form>
    )
  }
}
