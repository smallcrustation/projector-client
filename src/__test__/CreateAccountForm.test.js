import React from 'react'
import ReactDOM from 'react-dom'

import CreateAccountForm from '../components/CreateAccountForm/CreateAccountForm'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<CreateAccountForm />, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(<CreateAccountForm />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
