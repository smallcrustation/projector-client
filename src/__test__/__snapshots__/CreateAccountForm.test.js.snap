// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<form
  className="CreateAccountForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div>
     
    <div>
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div>
      <label
        htmlFor="pass"
      >
        Password
      </label>
      <input
        id="pass"
        name="pass"
        required={true}
        type="password"
      />
    </div>
    <div>
      <label
        htmlFor="pass2"
      >
        Confirm Password
      </label>
      <input
        id="pass2"
        name="pass2"
        required={true}
        type="password"
      />
    </div>
    <input
      type="submit"
      value="Create Account!"
    />
  </div>
</form>
`;
